### PR TITLE
User input producer up to #114

### LINKED ISSUE #97: Define precisely the functionalities expected from the user input producer
First step of #20 : I think before starting developing it the component we need to define precisely what is expected from the User Input producer component.

### PR DESCRIPTION
**Pull request by gilbrechbuhler**
Not finished yet but I do a pull request so everyone can review what has already been done.

Fix #97 #98 #101 #112 #113 #114

### CODE REVIEW

**Comment by gilbrechbuhler** [comment]
So last push is quite big. Some changes happened. The time base and all related options are now to be set when creating a new sequence. I did that because otherwise it was possible to create wrong sequences (documents with different time bases in the same sequences for example).

You can now choose between sending documents live when clicking the Send button, or schedule document sending. When the schedule is set on "Local clock" it uses the computer's clock, when it is set on "media" it uses the local clock with an offset. So when in media mode, you have to click on "Synchronize" button before sending documents to synchronize the "00:00:00" of the media clock to the current time of the computer. Then, still in media mode, if you set "00:00:10" as a scheduled time, it will send them ten seconds after the synchronization.

You can also send documents asynchronously now, you can set the number of seconds between each send.

Please review and comment, I know the html file is big, but actually most of the javascript is there for the interface : showing fields x and y when option z is selected, ... .

It certainly contains bugs, I tested some scenarios and did not discover bugs in this version, yet.

**Comment by gilbrechbuhler** [comment]
Now you can see the media clock running live once the synchronization is done.

**Comment by frans-ebu** [comment]
@gilbrechbuhler Many tnx for this piece of work! I am happy to see this merged.

One minor comment: I think it would be useful to add the URL of the input-producer to the readme, so a naive user can start to use it immediately.

**Comment by gilbrechbuhler** [comment]
@frans-ebu Done, good remark. I added this line to the README (in the Scripts part) : " To run it, just run `ebu-user-input-consumer` and open the file `ebu_tt_live/ui/user_input_producer/user_input_producer.html file.`"

**Comment by gilbrechbuhler** [comment]
I updated the documentation a bit also, to explain the new ways of send documents (scheduled time, asynchronous)

File: README.md [hunk 1]
```diff
@@ -32,7 +32,9 @@ and assumes the environment variables that the activation script sets):
 
     source ./env/bin/activate
 
-After having created the python virtual environment and having activated it the package
+To build the project you will also need node.js. Please read the instructions for your system [here](https://nodejs.org/en/download/package-manager/).
```
Comment (nigelmegitt) [inline README.md]: Is this accurate? I don't have node.js installed but I do have npm - they seem somewhat independent?

**Comment by nigelmegitt** [comment]
Review comments from meeting:
@kozmaz87 : uip is good to go
Me: this is good to merge, with the one minor query about node.js vs npm in the documentation.

Plus @EyalLavi has raised #122.

File: README.md (hunk 1 above)
Comment (gilbrechbuhler) [inline README.md]: For me npm was installed as a part of Node.js. Do you know a way to install npm only ? Because I did not find if there is.

Comment (gilbrechbuhler) [inline README.md]: Forget previous remark, we need Node.js. We use it to have npm but the scripts that precompiles the templates needs Node.js on the system anyway.

Comment (nigelmegitt) [inline README.md]: ok I must be misremembering - I thought I might have done something like `brew install npm` but when I do `brew list` it tells me I have node but not npm. Sorry about that.

**Comment by nigelmegitt** [comment]
Also #124, which is now fixed too.

**Pull request merged**